pkg/build/controller/controller.go

```diff
@@ -190,6 +168,48 @@ func (bc *BuildController) nextBuildPhase(build *buildapi.Build) error {
 	return nil
 }
 
+// resolveOutputDockerImageReference returns a reference to a Docker image
+// computed from the buid.Spec.Output.To reference.
+func (bc *BuildController) resolveOutputDockerImageReference(build *buildapi.Build) (string, error) {
+	outputTo := build.Spec.Output.To
+	if outputTo == nil || outputTo.Name == "" {
+		return "", nil
+	}
+	var ref string
+	switch outputTo.Kind {
+	case "DockerImage":
+		ref = outputTo.Name
+	case "ImageStream", "ImageStreamTag":
+		// TODO(smarterclayton): security, ensure that the reference image stream is actually visible
```

Comment:
if the imagestream/tag can't be resolved due to security or any other reason (non-existence) that should be handled in general by the retry logic (not seeing it is, but it should be) and we should be updating the build status with information about what the build is stuck on (there is an open issue about that general topic)